equinix/helpers_device.go

```diff
@@ -68,64 +69,46 @@ type NetworkInfo struct {
 	PrivateIPv4    string
 }
 
-func getNetworkInfoMetalGo(ips []metalv1.IPAssignment) NetworkInfo {
+func getNetworkInfo(ips []metalv1.IPAssignment) NetworkInfo {
 	ni := NetworkInfo{Networks: make([]map[string]interface{}, 0, 1)}
 	for _, ip := range ips {
 		network := map[string]interface{}{
-			"address": ip.Address,
-			"gateway": ip.Gateway,
-			"family":  ip.AddressFamily,
-			"cidr":    ip.Cidr,
-			"public":  ip.Public,
+			"address": ip.GetAddress(),
+			"gateway": ip.GetGateway(),
+			"family":  ip.GetAddressFamily(),
+			"cidr":    ip.GetCidr(),
+			"public":  ip.GetPublic(),
 		}
 		ni.Networks = append(ni.Networks, network)
 
 		// Initial device IPs are fixed and marked as "Management"
-		if *ip.Management {
-			if *ip.AddressFamily == 4 {
-				if *ip.Public {
-					ni.Host = *ip.Address
-					ni.IPv4SubnetSize = int(*ip.Cidr)
-					ni.PublicIPv4 = *ip.Address
+		if ip.GetManagement() {
```

Comment:
This is replacing pointer dereferences with `Get...()` calls for better protection against nil pointer errors.